src/native/linux_wayland.rs

```diff
@@ -533,14 +533,22 @@ impl crate::native::Clipboard for WaylandClipboard {
     fn set(&mut self, _data: &str) {}
 }
 
+
+pub fn eprint_iferr<T>(lib_name: &'static str, opt_lib: Option<T>) -> Option<T> {
+    if opt_lib.is_none() {
+        eprintln!("failed to load library {}", lib_name);
+    }
+    return opt_lib
+}
+
 pub fn run<F>(conf: &crate::conf::Conf, f: &mut Option<F>) -> Option<()>
 where
     F: 'static + FnOnce() -> Box<dyn EventHandler>,
 {
     unsafe {
-        let client = LibWaylandClient::try_load()?;
-        let egl = LibWaylandEgl::try_load()?;
-        let xkb = LibXkbCommon::try_load()?;
+        let client = eprint_iferr("libwayland-client", LibWaylandClient::try_load())?;
```

Comment:
Maybe try_load could return a better message instead of handling this here?

Reply:
for sure, it would be more natural for the various Lib___::try_load() functions to return Result<>s rather than Option<>s